src/lib/v1/user/validator.ts

```diff
@@ -0,0 +1,49 @@
+import Joi from 'joi';
+
+import { config } from '../../../config';
+import { createRouteParams, listRouteParams, patchRouteParams, updateRouteParams } from './types';
+
+export const validatorPatchRoute = (params: patchRouteParams): Joi.ValidationResult => {
```

Comment:
since Joi library itself is not typed, declare type for `params` doesn't help much, you can't prevent one day `params` and joi schema out of sync.

zod is better solution: https://github.com/colinhacks/zod

zod can infer type from schema, so you can directly pass the validated data to `patchUserById` function, if the inferred type doesn't match with `patchRouteParams`, compiler can warn you.

Reply:
Oh, good call I didn't know this library.
Will be handled in Issue https://github.com/Faz95210/test-backend/issues/11

Reply:
just a suggestion, it's not essential

> Oh, good call I didn't know this library. Will be handled in Issue [Faz95210#11](https://github.com/Faz95210/test-backend/issues/11)

Reply:
Oh I know, I just figured since I never used zod before it would be a good opportunity to try something new.

After trying zod I do feel I prefer using Joi for small project though.